torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -546,15 +546,15 @@ struct NCCLTraceBuffer {
 
     std::unique_lock<std::mutex> guard(mutex_);
 
-    auto& entry = entries_.at(*id % max_entries_);
-    if (entry.id_ == *id) {
-      update_state(entry);
+    Entry* entry = &entries_.at(*id % max_entries_);
```

Comment:
Not sure if I missed anything, but it seems that the code would behave identically before and after the PR? https://godbolt.org/z/49Pn89aaG